akka-http/src/main/scala/akka/http/scaladsl/server/util/JwtSupport.scala

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2022 Lightbend Inc.
+ */
+
+package akka.http.scaladsl.server.util
+
+import java.io.File
+import java.security.KeyPair
+import java.util.Base64
+
+import com.typesafe.config.Config
+import javax.crypto.SecretKey
+import javax.crypto.spec.SecretKeySpec
+import pdi.jwt.{ algorithms, JwtAlgorithm, JwtOptions }
+import spray.json.{ JsObject, JsString }
+
+import scala.util.{ Failure, Success, Try }
+import scala.jdk.CollectionConverters._
+
+trait JwtSupport {
+
+  def canValidate: Boolean
+
+  def validate(token: String): Either[Exception, JsObject]
+
+}
+
+object JwtSupport {
+
+  private val NoValidationOptions = JwtOptions.DEFAULT.copy(signature = false, expiration = false, notBefore = false)
+
+  def fromConfig(config: Config): JwtSupport = {
```

Comment:
I think this should all go in the `JwtSettings` and be loaded through `SettingsCompanionImpl#apply(ActorSystem)` by default, since that caches the instance rather than re-parses all config/reloads key in this case for every incoming request.

Especially important since it does blocking read of keypair from filesystem.

Reply:
thanks. done in https://github.com/akka/akka-http/pull/4377/commits/c02d9ba4a853d4a5bdb952614c0bebe813d7d0dd